extensions/integration-tests/src/utils.ts

```diff
@@ -137,10 +138,63 @@ export async function assertFileGenerationResult(filepath: string, retryCount: n
 	while (retryCount > 0 && !exists) {
 		--retryCount;
 		exists = fs.existsSync(filepath);
-		await sleep(5000);
+
+		if (!exists) {
+			await sleep(5000);
+		}
 	}
 
 	assert(exists, `File ${filepath} is expected to be present`);
 	assert(fs.readFileSync(filepath).byteLength > 0, 'File should not be empty');
 	fs.unlinkSync(filepath);
-}
\ No newline at end of file
+}
+
+/**
+ *
+ * @param databaseName name of database where to look for table
+ * @param tableName table to look for
+ * @param ownerUri owner uri
+ * @param retryCount number of times to retry with a 5 second wait between each try
+ * Checks for table existing
+ */
+export async function assertTableCreationResult(databaseName: string, tableName: string, ownerUri: string, retryCount: number, checkForData?: boolean): Promise<void> {
+	let result: azdata.SimpleExecuteResult;
+	while (retryCount > 0) {
+		--retryCount;
+		let query = `BEGIN TRY
+				USE ${databaseName}
+				SELECT * FROM INFORMATION_SCHEMA.TABLES WHERE TABLE_NAME = '${tableName}'
+			END TRY
+			BEGIN CATCH
+				SELECT ERROR_MESSAGE() AS ErrorMessage;
+			END CATCH`;
+		result = await runQuery(query, ownerUri);
+		if (result.rowCount > 0) {
+			break;
+		}
+		await sleep(5000);
+	}
+
+	assert(result.rowCount === 1, `Table ${tableName} should be created`);
```

Comment:
If there's >1 table returned here this will fail (see comment above). Fixing that should make this a non-issue but I'd suggest outputting the number of rows as well just to make debugging easier